src/adapters/nats.js

```diff
@@ -0,0 +1,436 @@
+					message.working();
+					await chan.handler(this.serializer.deserialize(message.data), message);
+					message.ack();
+				} catch (error) {
+					// this.logger.error(error);
+
+					// Message rejected
+					if (!chan.maxRetries) {
+						// No retries
+
+						if (chan.deadLettering.enabled) {
+							this.logger.debug(
+								`No retries, moving message to '${chan.deadLettering.queueName}' queue...`
+							);
+							await this.moveToDeadLetter(chan, message);
+						} else {
+							// Drop message
+							this.logger.error(`No retries, drop message...`, message.seq);
+						}
+
+						message.ack();
+					} else if (
+						chan.maxRetries > 0 &&
+						message.info.redeliveryCount >= chan.maxRetries
+					) {
+						// Retries enabled and limit reached
+
+						if (chan.deadLettering.enabled) {
+							this.logger.debug(
+								`Message redelivered too many times (${message.info.redeliveryCount}). Moving message to '${chan.deadLettering.queueName}' queue...`
+							);
+							await this.moveToDeadLetter(chan, message);
+						} else {
+							// Drop message
+							this.logger.error(
+								`Message redelivered too many times (${message.info.redeliveryCount}). Drop message...`,
+								message.seq
+							);
+							// this.logger.error(`Drop message...`, message.seq);
+						}
+
+						message.ack();
+					} else {
+						// Retries enabled but limit NOT reached
+						// NACK the message for redelivery
+						this.logger.debug(`NACKing message...`, message.seq);
+						message.nak();
+					}
+				}
+
+				this.removeChannelActiveMessages(chan.id, [message.seq]);
+			}
+		};
+	}
+
+	/**
+	 * Create a NATS Stream
+	 *
+	 * More info: https://docs.nats.io/jetstream/concepts/streams
+	 *
+	 * @param {String} streamName Name of the Stream
+	 * @param {Array<String>} subjects A list of subjects/topics to store in a stream
+	 * @param {StreamConfig} streamOpts JetStream stream configs
+	 */
+	async createStream(streamName, subjects, streamOpts) {
+		const streamConfig = _.defaultsDeep(
+			{
+				name: streamName,
+				subjects: subjects
+			},
+			streamOpts,
+			this.opts.nats.streamConfig
+		);
+
+		try {
+			const streamInfo = await this.manager.streams.add(streamConfig);
+			this.logger.debug(streamInfo);
+		} catch (error) {
+			if (error.message === "stream name already in use") {
+				// Silently ignore the error. Channel or Consumer Group already exists
+				this.logger.debug(`NATS Stream with name: '${streamName}' already exists.`);
+			} else {
+				this.logger.error("An error ocurred while create NATS Stream", error);
+			}
+		}
+	}
+
+	/**
+	 * Moves message into dead letter
+	 *
+	 * @param {Channel} chan
+	 * @param {JsMsg} message JetStream message
+	 */
+	async moveToDeadLetter(chan, message) {
+		// this.logger.warn(`Moved message to '${chan.deadLettering.queueName}'`);
+
+		try {
+			await this.publish(chan.deadLettering.queueName, message.data, { raw: true });
```

Comment:
Please add info about the original channel. E.g AMQP adapter sets `x-original-channel` header.